integrations/phishing_analyzers/analyzers/driver_wrapper.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import os
+from typing import Iterator
+
+from selenium.common import WebDriverException
+from seleniumwire.request import Request
+from seleniumwire.webdriver import ChromeOptions, Remote
+
+LOG_NAME = "driver_wrapper"
+
+# get flask-shell2http logger instance
+logger = logging.getLogger(LOG_NAME)
+# logger config
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+log_level = os.getenv("LOG_LEVEL", logging.INFO)
+log_path = os.getenv("LOG_PATH", f"/var/log/intel_owl/{LOG_NAME}")
+# create new file handlers, files are created if doesn't already exists
+fh = logging.FileHandler(f"{log_path}/{LOG_NAME}.log")
+fh.setFormatter(formatter)
+fh.setLevel(log_level)
+fh_err = logging.FileHandler(f"{log_path}/{LOG_NAME}_errors.log")
+fh_err.setFormatter(formatter)
+fh_err.setLevel(logging.ERROR)
+# add the handlers to the logger
+logger.addHandler(fh)
+logger.addHandler(fh_err)
+logger.setLevel(log_level)
+
+
+class DriverWrapper:
+    def __init__(
+        self,
+        proxy_address: str = "",
+        window_width: int = 1920,
+        window_height: int = 1080,
+    ):
+        self.proxy: str = proxy_address
+        self.window_width: int = window_width
+        self.window_height: int = window_height
+        self.last_url: str = ""
+        self._driver: Remote = self._init_driver(self.window_width, self.window_height)
+
+    def _init_driver(self, window_width: int, window_height: int) -> Remote:
+        logger.info(f"Adding proxy with option: {self.proxy}")
+        logger.info("Creating Chrome driver...")
```

Comment:
I will add it to the `navigate()` method since it's where it will look for the URL, if it sounds right to you

Reply:
I would add the URL anywhere is possible. The use case is looking for that URL in Elastic logs. If you don't have this information, you cannot track all the steps for a specific analysis and these logs would be pretty useless because they cannot be related to any action